### ISSUE TITLE
Publish the third aar dependencies

### ISSUE THREAD

**Issue by Sugerlad** [sky-uk/gradle-maven-plugin#14]
In my library module. I add a dependence as [it says](https://github.com/sinaweibosdk/weibo_android_sdk):
```
allprojects {
    repositories {
        jcenter()
        mavenCentral()
        maven { url "https://dl.bintray.com/thelasterstar/maven/" }
    }
}
compile 'com.sina.weibo.sdk:core:4.3.8:openDefaultRelease@aar'
```
__NOTICE__: The `coordinate` with `openDefaultRelease` part.

Also I try the other method:
```
repositories{
    flatDir {
        dirs 'libs'
    }
}

dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    compile 'com.android.support:appcompat-v7:24.2.1'
    compile(name: 'openDefault-4.3.9', ext: 'aar')

}
```
But when I run `./gradlew publish` the both don't containe the dependence.

**Comment by gotev** [comment]
@Sugerlad if I understood correctly, you have a library which depends on `com.sina.weibo.sdk:core:4.3.8` and you want your final aar to contain their aar too?

**Comment by Sugerlad** [comment]
@gotev Yes, that's right. I want use the `library` to my other project. But after I add the library into the project, the error occures that the class not found contained by  `com.sina.weibo.sdk:core:4.3.8`.

**Comment by gotev** [comment]
This is not something you could accomplish with this plugin. It's not meant to make a "fat library" made of your library plus all your dependencies bundled together, but to create your aar and describe its dependencies. To accomplish what you are asking, you need something like the [shadow plugin](https://github.com/johnrengelman/shadow) instead, but I strongly recommend you to not do it and simply inform your library users that they need to add:

```
maven { url "https://dl.bintray.com/thelasterstar/maven/" }
```

as well. Always make dependencies explicit. It eases maintenance and ensures your users are aware of what they are importing. This is crucial also for government laws such as the european GDPR and license statements.

**Comment by Sugerlad** [comment]
Sorry @gotev, I have not explained clearly my problem before. In my `library` I add a dependency `compile 'com.sina.weibo.sdk:core:4.3.8:openDefaultRelease@aar'`. Now I want publish the `library` for other project used. Because of the dependency classifier `openDefaultRelease`, the pom generated by your plugin does not include `classifier` part. When I add the `library` in other project, it can not find the `com.sina.weibo.sdk:core:4.3.8:openDefaultRelease@aa`. I think the reason is that the pom does not include `openDefaultRelease` classifier.

**Comment by Fed93** [comment]
@Sugerlad I'm investigating on this issue.

**Comment by Fed93** [comment]
@Sugerlad i've added the classifier handling in the plugin, can you try to publish your library using the plugin from the branch `add-classifier` using 
`apply from: "https://raw.githubusercontent.com/sky-uk/gradle-maven-plugin/${project.mavPluginVersion}/gradle-mavenizer.gradle"` and adding `mavPluginVersion = 'add-classifier'` in your plugin configuration?

**Comment by Sugerlad** [comment]
Thanks @Fed93 . I create a new project [here](https://github.com/Sugerlad/IdleLibrary) and use the branch `add-classifier`. Jitpack builds success.The log is [here](https://jitpack.io/com/github/Sugerlad/IdleLibrary/0.0.1/build.log) and [here](https://jitpack.io/#Sugerlad/IdleLibrary/0.0.1) . I add the dependency `implementation 'com.github.Sugerlad:IdleLibrary:0.0.1'` and run `./gradle build`. The error occurs as the same before:
```gradle
> Could not resolve all files for configuration ':app:debugCompileClasspath'.
   > Could not find core.jar (com.sina.weibo.sdk:core:4.4.0).
     Searched in the following locations:
         https://dl.bintray.com/thelasterstar/maven/com/sina/weibo/sdk/core/4.4.0/core-4.4.0.jar


```

**Comment by Fed93** [comment]
@Sugerlad can you try to rebuild your library now, using the same branch?
I've added the extension of the library in the pom file.

**Comment by Sugerlad** [comment]
Thanks @Fed93. I use the same `add-classifier ` branch and run `./gradlew publish`. The generated pom as follows(part):
```
<dependencies>
    <dependency>
      <groupId>com.sina.weibo.sdk</groupId>
      <artifactId>core</artifactId>
      <version>4.4.0</version>
      <scope>compile</scope>
      <exclusions>
        <exclusion>
          <artifactId>*</artifactId>
          <groupId>*</groupId>
        </exclusion>
      </exclusions>
    </dependency>
   ....//ignore
  </dependencies>
```
The classifier not been included. The relative dependency is 
`api 'com.sina.weibo.sdk:core:4.4.0:openDefaultRelease@aar'`

**Comment by skymontif** [comment]
@Sugerlad please check the apply path of the plugin, it must be `apply from: "https://raw.githubusercontent.com/sky-uk/gradle-maven-plugin/${project.mavPluginVersion}/gradle-mavenizer.gradle"` with `mavPluginVersion = 'add-classifier'` in your project ext and make sure to Sync Gradle.

**Comment by Fed93** [comment]
@Sugerlad please check the apply path of the plugin, it must be `apply from: "https://raw.githubusercontent.com/sky-uk/gradle-maven-plugin/${project.mavPluginVersion}/gradle-mavenizer.gradle"` with `mavPluginVersion = 'add-classifier'` in your project ext and make sure to Sync Gradle.

**Comment by Sugerlad** [comment]
Sorry @Fed93 , I missed `mavPluginVersion = 'add-classifier'` in the project build file before. Now everything is OK. You can close this issue now. Thanks @Fed93 .

**Comment by Fed93** [comment]
@Sugerlad when new version of the plugin is published i'll close the issue.
Please stay tuned and revert your apply path to master when update is out in order to use the latest version of the plugin!
